src/envoy/utils/utils_test.cc

```diff
@@ -44,4 +44,31 @@ TEST(UtilsTest, ParseMessageWithUnknownField) {
   EXPECT_EQ(http_config.default_destination_service(),
             "service.svc.cluster.local");
 }
+
+TEST(UtilsTest, GetSourceNamespace) {
```

Comment:
Can you add test cases where ns/ is not the start of the principal.

Reply:
Done.